validation-test/compiler_crashers_2_fixed/0149-rdar38694450.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-swift-frontend %s -emit-ir -o /dev/null
```

Comment:
A better test would go in test/SILOptimizer and use FileCheck to ensure the cast was not optimized away

Reply:
That's doesn't quite cover the symptoms, since the crash was in IRGen, but I'll FileCheck in addition.